public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -262,7 +271,7 @@ export class PanelModel {
     const reactPanel = newPlugin.exports.reactPanel;
 
     // for angular panels we must remove all events and let angular panels do some cleanup
-    if (!reactPanel) {
+    if (!_.has(this, 'plugin.exports.reactPanel')) {
```

Comment:
ah, good fix! but really not loving these lodash get/has, now that we have such fine typing it's a shame to use things like this.

Reply:
is:
```
if (!(this.plugin && this.plugin.exports &&  this.plugin.exports && this.plugin.exports.reactPanel) 
```
better?  😱 

Is there a good typed way to check these chains of possible null/missing values?